### PR TITLE
fix(versions): show version uploaded by File Request

### PR DESCRIPTION
**Pull request by omgrief**

### CODE REVIEW

File: src/elements/common/selectors/version.js [hunk 1]
```diff
@@ -32,8 +34,15 @@ const getVersionAction = ({ restored_at, trashed_at, version_promoted }: $Shape<
     return action;
 };
 
-const getVersionUser = ({ modified_by, restored_by, trashed_by }: $Shape<BoxItemVersion>): User => {
-    return restored_by || trashed_by || modified_by || PLACEHOLDER_USER;
+const getVersionUser = (
+    { modified_by, restored_by, trashed_by, uploader_display_name }: $Shape<BoxItemVersion>,
+    intl: IntlShape,
+): User => {
+    if (restored_by || trashed_by) return restored_by || trashed_by;
```
Comment (jstoffan) [inline src/elements/common/selectors/version.js:41]: I find this to be very difficult to read. We usually avoid inline conditional returns.

File: src/elements/common/selectors/version.js [hunk 2]
```diff
@@ -32,8 +34,15 @@ const getVersionAction = ({ restored_at, trashed_at, version_promoted }: $Shape<
...
+    if (modified_by && modified_by.id === PLACEHOLDER_USER.id && uploader_display_name) {
+        return { ...modified_by, name: internationalizeDisplayName(uploader_display_name, intl) };
```
Comment (jstoffan) [inline src/elements/common/selectors/version.js:43]: Selectors should ideally not be aware of user-facing text or internationalization. Can we handle this in a presentational component elsewhere?

File: src/elements/common/selectors/version.js [hunk 3]
```diff
@@ -32,8 +34,15 @@ const getVersionAction = ({ restored_at, trashed_at, version_promoted }: $Shape<
...
 };
 
-const getVersionUser = ({ modified_by, restored_by, trashed_by }: $Shape<BoxItemVersion>): User => {
-    return restored_by || trashed_by || modified_by || PLACEHOLDER_USER;
+const getVersionUser = (
+    { modified_by, restored_by, trashed_by, uploader_display_name }: $Shape<BoxItemVersion>,
+    intl: IntlShape,
+): User => {
+    if (restored_by || trashed_by) return restored_by || trashed_by;
+    if (modified_by && modified_by.id === PLACEHOLDER_USER.id && uploader_display_name) {
```
Comment (jstoffan) [inline src/elements/common/selectors/version.js:42]: I think `modified_by === PLACEHOLDER_USER` would work as a simplified version of this condition.

File: src/elements/content-sidebar/activity-feed/version/Version.js [hunk 4]
```diff
@@ -40,7 +40,7 @@ const Version = (props: Props): React.Node => {
     const action = selectors.getVersionAction(props);
     const { id, intl, onInfo, version_number, version_promoted } = props;
     // $FlowFixMe
-    const { name } = selectors.getVersionUser(props);
+    const { name } = selectors.getVersionUser(props, intl);
```
Comment (jstoffan) [inline src/elements/content-sidebar/activity-feed/version/Version.js:43]: I believe `name` is empty for PLACEHOLDER_USER. Can we do something like the following?

```
const { name } = selectors.getVersionUser(props);
const uploaderName = uploader_display_name === FILE_REQUEST ? intl.formatMessage(messages.displayName) : uploader_display_name;
const displayName = name || uploaderName || '';
```

File: src/elements/content-sidebar/versions/__tests__/VersionsItem.test.js [hunk 5]
```diff
@@ -3,7 +3,7 @@ import { FormattedMessage } from 'react-intl';
 import { shallow } from 'enzyme/build';
 import messages from '../messages';
 import selectors from '../../../common/selectors/version';
-import VersionsItem from '../VersionsItem';
+import { VersionsItem } from '../VersionsItem';
```
Comment (jstoffan) [inline src/elements/content-sidebar/versions/__tests__/VersionsItem.test.js:6]: What is the reason for this change?

File: src/elements/content-sidebar/versions/VersionsItem.js [hunk 6]
```diff
@@ -82,7 +84,7 @@ const VersionsItem = ({
     const versionInteger = versionNumber ? parseInt(versionNumber, 10) : 1;
     const versionTime = restoredAt || trashedAt || createdAt;
     const versionTimestamp = versionTime && new Date(versionTime).getTime();
-    const versionUserName = selectors.getVersionUser(version).name || (
+    const versionUserName = selectors.getVersionUser(version, intl).name || (
```
Comment (jstoffan) [inline src/elements/content-sidebar/versions/VersionsItem.js:87]: Can we use `FormattedMessage` to avoid injecting `intl`?

File: src/elements/common/selectors/__tests__/version.js [hunk 7]
```diff
@@ -7,12 +7,19 @@ import {
     VERSION_UPLOAD_ACTION,
 } from '../../../../constants';
 
+import { FILE_REQUEST } from '../../../../features/file-request/constants';
+import messages from '../../../../features/file-request/messages';
```
Comment (jstoffan) [inline src/elements/common/selectors/__tests__/version.js:11]: We should avoid importing feature-specific modules into a common module, if possible.

File: src/elements/content-sidebar/versions/__tests__/VersionsItem.test.js (hunk 5 above)
Comment (jstoffan) [reply src/elements/content-sidebar/versions/__tests__/VersionsItem.test.js:6]: I saw after the fact that we've introduced a HOC to the component. Disregard.

File: src/utils/fields.js [hunk 8]
```diff
@@ -134,6 +134,7 @@ const FILE_VERSION_FIELDS_TO_FETCH = [
     FIELD_RESTORED_FROM,
     FIELD_SIZE,
     FIELD_VERSION_NUMBER,
+    FIELD_UPLOADER_DISPLAY_NAME,
```
Comment (jstoffan) [inline src/utils/fields.js:137]: Please alphabetize all declarations, imports, object properties, etc. when possible.

File: src/elements/content-sidebar/versions/VersionsItem.js [hunk 9]
```diff
@@ -82,9 +85,18 @@ const VersionsItem = ({
     const versionInteger = versionNumber ? parseInt(versionNumber, 10) : 1;
     const versionTime = restoredAt || trashedAt || createdAt;
     const versionTimestamp = versionTime && new Date(versionTime).getTime();
-    const versionUserName = selectors.getVersionUser(version).name || (
-        <FormattedMessage {...messages.versionUserUnknown} />
-    );
+    const versionUser = selectors.getVersionUser(version);
+    let versionUserName = versionUser.name || <FormattedMessage {...messages.versionUserUnknown} />;
+    if (uploader_display_name) {
+        if (versionUser.id === PLACEHOLDER_USER.id) {
```
Comment (jstoffan) [inline src/elements/content-sidebar/versions/VersionsItem.js:91]: Do these conditions need to be nested or can we make judicious use of `&&`?

File: src/elements/content-sidebar/versions/VersionsItem.js [hunk 10]
```diff
@@ -72,6 +74,7 @@ const VersionsItem = ({
         trashed_at: trashedAt,
         version_number: versionNumber,
         version_promoted: versionPromoted,
+        uploader_display_name,
```
Comment (jstoffan) [inline src/elements/content-sidebar/versions/VersionsItem.js:77]: Let's rename to use camelCase to match the conventions of the file.

File: src/elements/content-sidebar/activity-feed/version/Version.js [hunk 11]
```diff
@@ -38,17 +41,25 @@ const ACTION_MAP = {
 const Version = (props: Props): React.Node => {
     // $FlowFixMe
     const action = selectors.getVersionAction(props);
-    const { id, intl, onInfo, version_number, version_promoted } = props;
+    const { id, intl, onInfo, version_number, version_promoted, uploader_display_name } = props;
     // $FlowFixMe
-    const { name } = selectors.getVersionUser(props);
-
+    const user = selectors.getVersionUser(props);
+    let display_name = user.name;
```
Comment (jstoffan) [inline src/elements/content-sidebar/activity-feed/version/Version.js:47]: We prefer to use destructuring to rename properties rather than rename-only reassignments. We should also use camelCase for all variable names.

File: src/constants.js [hunk 12]
```diff
@@ -381,6 +381,7 @@ export const VERSION_RETENTION_INDEFINITE: 'indefinite' = 'indefinite';
 
 /* ------------------ Placeholder Feed Items ------------------------- */
 export const PLACEHOLDER_USER = { type: 'user', id: '2', name: '' };
+export const FILE_REQUEST = 'File Request';
```
Comment (jstoffan) [inline src/constants.js:384]: Is this meant to be equivalent a user name? Can we call it `FILE_REQUEST_NAME` to indicate that?

File: src/elements/content-sidebar/versions/messages.js [hunk 13]
```diff
@@ -3,6 +3,11 @@
 import { defineMessages } from 'react-intl';
 
 const messages = defineMessages({
+    fileRequestDisplayName: {
+        defaultMessage: 'File Request',
+        description: 'File Request feature display name',
```
Comment (jstoffan) [inline src/elements/content-sidebar/versions/messages.js:8]: We may want to provide a bit more context so the translator can provide a meaningful translation.

File: src/elements/content-sidebar/versions/VersionsItem.js [hunk 14]
```diff
@@ -82,9 +85,18 @@ const VersionsItem = ({
...
     const versionTimestamp = versionTime && new Date(versionTime).getTime();
-    const versionUserName = selectors.getVersionUser(version).name || (
-        <FormattedMessage {...messages.versionUserUnknown} />
-    );
+    const versionUser = selectors.getVersionUser(version);
+    let versionUserName = versionUser.name || <FormattedMessage {...messages.versionUserUnknown} />;
+    if (uploader_display_name) {
+        if (versionUser.id === PLACEHOLDER_USER.id) {
+            versionUserName =
+                uploader_display_name === FILE_REQUEST ? (
```
Comment (jstoffan) [inline src/elements/content-sidebar/versions/VersionsItem.js:93]: I'm seeing some repetition between components. It seems like we're trying to do two things: 

1. Determine if the user is anonymous and, if so, substitute the blank name for `uploader_display_name` if available
2. Translate the `uploader_display_name` if it's a generic server-generated value (I assume)

It seems like the code for goal 1 should live in `selectors.getVersionUser` with a small helper method to extend/modify PLACEHOLDER_USER as needed.

It seems like the code for goal 2 should live in each component, since they each use different implementations to internationalize content (`injectIntl` vs `FormattedMessage`).

File: src/elements/content-sidebar/versions/VersionsItem.js (hunk 9 above)
Comment (omgrief) [reply src/elements/content-sidebar/versions/VersionsItem.js:91]: ooh yes, that can probably happen

File: src/elements/content-sidebar/activity-feed/version/__tests__/Version.test.js [hunk 15]
```diff
@@ -2,10 +2,11 @@ import * as React from 'react';
 import { shallow } from 'enzyme';
 import selectors from '../../../../common/selectors/version';
 import { VersionBase as Version } from '../Version';
-import { PLACEHOLDER_USER, VERSION_UPLOAD_ACTION } from '../../../../../constants';
+import { FILE_REQUEST_NAME, PLACEHOLDER_USER, VERSION_UPLOAD_ACTION } from '../../../../../constants';
+import messages from '../../../../common/messages';
 
 const translationProps = {
-    intl: { formatMessage: () => {} },
+    intl: { formatMessage: anyString => anyString },
```
Comment (sahiga) [inline src/elements/content-sidebar/activity-feed/version/__tests__/Version.test.js:9]: Where is `anyString` defined?

Comment (omgrief) [reply src/elements/content-sidebar/activity-feed/version/__tests__/Version.test.js:9]: it's just a function argument that makes formatMessage return whatever is passed in
